frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx

```diff
@@ -181,12 +251,31 @@ class ExpressionEditorTextfield extends React.Component<
     }
   }
 
+  handleKeypress = (evt: KeyboardEvent) => {
+    if (evt.key !== "Enter") {
+      return;
+    }
+
+    evt.preventDefault();
+    evt.stopPropagation();
+    this.handleEnter();
+  };
+
+  textarea() {
+    return this.input.current?.refEditor?.getElementsByTagName("textarea")[0];
+  }
+
   componentDidMount() {
     if (this.input.current) {
       const { editor } = this.input.current;
       // "ExpressionMode" constructor is not typed, so cast it here explicitly
       const mode = new ExpressionMode() as unknown as Ace.SyntaxMode;
 
+      // HACK: manually register the keypress event for the enter key,
+      // since ACE doesn ot seem to call the event handlers in time for
+      // them to do certain things, like window.open.
+      this.textarea()?.addEventListener("keypress", this.handleKeypress);
```

Comment:
ACE editor does not call the keypress handler in time for it to open a link (popups get blocked if they are not in response to a user action), so we need to listen to this keypress manually.

Reply:
> popups get blocked if they are not in response to a user action

This would be useful in the in-code comment as well :+1: